### ISSUE TITLE
Implement lock operations

### ISSUE THREAD

**Issue by hajnalandor** [PumpkinSeed/bucket#62]

**Comment by PumpkinSeed** [comment]
Can you tell more details about what you mean?

**Comment by hajnalandor** [comment]
In gocb package there are 2 methods, GetAndLock and Unlock. GetAndLock locks a document for a period of time, providing exclusive RW access to it,meanwhile Unlock works in the reverse direction.

**Comment by PumpkinSeed** [comment]
Yes, I used this GetAndLock in the incrmntr package, but it's strictly related to the #49, becase after you call GetAndLock the only way to release the lock if you use the Cas or without any operation call the Unlock.
